components/page-templates/B2B/UsersTemplate/UsersTemplate.tsx

```diff
@@ -1,18 +1,309 @@
+      ...paginationState,
+      startIndex: (data?.pageSize ?? 0) * (page - 1),
+    })
+
+  const handleAddUser = async (formValues: B2BUserInput) => {
+    const variables = buildCreateCustomerB2bUserParams({ user, values: formValues })
+    const createUserResponse = await createCustomerB2bUser.mutateAsync({
+      ...variables,
+    })
+    if (createUserResponse?.userId) {
+      addRoleToB2bUser(createUserResponse, formValues)
+    }
+  }
+
+  const handleUpdateUser = async (formValues: B2BUserInput, b2BUser?: B2BUser | undefined) => {
+    const variables = buildUpdateCustomerB2bUserParams({ user, b2BUser, values: formValues })
+    const updateUserResponse = await updateCustomerB2bUser.mutateAsync({
+      ...variables,
+    })
+    const previousRoles = b2BUser?.roles as CustomerB2BUserRole[]
+    if (
+      updateUserResponse &&
+      previousRoles &&
+      previousRoles.length &&
+      formValues.role !== previousRoles[0]?.roleName
+    ) {
+      await deleteB2bAccountUserRole.mutateAsync(
+        buildB2bUserRoleParams({
+          user,
+          b2BUser,
+          values: {
+            ...formValues,
+            role: previousRoles[0]?.roleName,
+          },
+          roles: userRoles,
+        })
+      )
+    }
+    addRoleToB2bUser(updateUserResponse, formValues)
+  }
+
+  const addRoleToB2bUser = async (b2BUser: B2BUser, formValues: any) => {
+    const addRoleToCustomerB2bAccountVariables = buildB2bUserRoleParams({
+      user,
+      b2BUser: b2BUser,
+      values: formValues,
+      roles: userRoles,
+    })
+    await addRoleToCustomerB2bAccount.mutateAsync({
+      ...addRoleToCustomerB2bAccountVariables,
+    })
+  }
+
+  // const handleAddUserDesktopButtonClick = () => {
+
+  // }
+
+  const handleAddUserButtonClick = () => {
+    if (mdScreen) {
+      setIsUserFormOpen(true)
+    } else {
+      showModal({
+        Component: UserFormDialog,
+        props: {
+          isEditMode: false,
+          isUserFormInDialog: true,
+          formTitle: t('add-new-user'),
+          b2BUser: undefined,
+          onSave: (b2BUserInput: B2BUserInput) => handleAddUser(b2BUserInput),
+          onClose: () => {
+            setIsUserFormOpen(false)
+            closeModal()
+          },
+        },
+      })
+    }
+  }
+
+  return (
+    <Grid>
+      <Grid item style={{ marginTop: '10px', marginBottom: '20px' }}>
+        <BackButtonLink aria-label={t('my-account')} href="/my-account">
+          <ChevronLeftIcon />
+          {mdScreen && <Typography variant="body1">{t('my-account')}</Typography>}
+        </BackButtonLink>
+        <Box sx={UsersTemplateStyle.heading}>
+          <Typography
+            variant={mdScreen ? 'h1' : 'h2'}
+            sx={{ textAlign: { xs: 'center' }, marginTop: { xs: '-3.6rem', md: 0 } }}
+          >
+            {t('users')}
+          </Typography>
+        </Box>
+        <Grid container>
+          <Grid item xs={12} md={12}>
+            <AddUserButton
```

Comment:
Don't create a separate component for this

Reply:
Done